### PR TITLE
Build and deploy

### PR DESCRIPTION
**Pull request by fesor**
- docker-compose files was DRY'd. Make use of variable substitution. 
 - Scripts updated for docker-compose 1.7. All default values for env variables are now stored in `.env` file. ([details](https://github.com/docker/compose/releases/tag/1.7.0))  
 - Reduced total size of incremental build size by splitting COPY process and placing vendors part up in Dockerfile. This will allow to make very tiny incremental build (by using cache for vendor code).
 - Add build script, which is building images and pushes them to the registry
 - Add deployment instruction. Currently I propose to use docker-machine. By doing this deployment is just as simple as this bash script:

```
eval $(docker-machine env your-remote-host)

COMPOSE_FILE=docker-compose.yml:docker-compose.prod.yml
docker-compose pull
docker-compose up -d
```

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -19,7 +19,14 @@ RUN yes | pecl install xdebug-beta \
         && echo "xdebug.remote_connect_back=on" >> /usr/local/etc/php/conf.d/xdebug.ini
 
 COPY support/php/fpm_www.conf /usr/local/etc/php-fpm.d/www.conf
-COPY . /srv/
+
+COPY vendor /srv/vendor
```
Comment (snsanich) [inline Dockerfile]: I suggest use `COPY` command only once.

File: support/scripts/build [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+docker run --rm \
+     --user $(id -u) \
+     -v $(pwd):/app \
+     -v ~/.composer:/composer \
+     fesor/composer install --ignore-platform-reqs --optimize-autoloader
+
+export COMPOSE_FILE=docker-compose.yml:docker-compose.prod.yml
+docker-compose build --force-rm
+# Spooky bash hack for getting all images built by docker-compose
+images=$(docker-compose config | grep image | sed -e 's/^[ \t]*image\:[ ]*//')
```
Comment (snsanich) [inline support/scripts/build]: Looks fascinating!

**Comment by snsanich** [comment]
Currently Changes are contradict with doc:
* At first you run export PATH=./support...
* Then you run composer install ...
* And then *in doc* it is noticed you should run compose up, but "compose" command does not exist.

Also When I ran docker-compose up as third (3) step, I got:

    WARNING: The DB_USER variable is not set. Defaulting to a blank string.
    WARNING: The DB_NAME variable is not set. Defaulting to a blank string.
    WARNING: The DB_PASS variable is not set. Defaulting to a blank string.
    WARNING: The APP_SECRET variable is not set. Defaulting to a blank string.

**Comment by fesor** [comment]
@snsanich run `docker-compose --version` and be sure that you are using 1.7+

**Pull request merged**